javascriptlib/src/mill/javascriptlib/TestModule.scala

```diff
@@ -25,6 +25,82 @@ trait TestModule extends TaskModule {
 object TestModule {
   type TestResult = Unit
 
+  trait Coverage extends TypeScriptModule with TestModule {
+    override def npmDevDeps: T[Seq[String]] = Task {
+      super.npmDevDeps() ++ Seq("serve@14.2.4")
+    }
+
+    private[TestModule] def runCoverage: T[TestResult]
+
+    protected def coverageTask(args: Task[Seq[String]]): Task[TestResult] = Task { runCoverage() }
+
+    def coverage(args: String*): Command[TestResult] =
+      Task.Command {
+        coverageTask(Task.Anon { args })()
+      }
+
+    // <rootDir> = '/out'; allow coverage resolve distributed source files.
+    // & define coverage files relative to <rootDir>.
+    private[TestModule] def link: Task[Unit] = Task.Anon {
+      os.symlink(Task.workspace / "out/node_modules", npmInstall().path / "node_modules")
+      os.symlink(Task.workspace / "out/tsconfig.json", compile()._1.path / "tsconfig.json")
+      if (os.exists(compile()._1.path / ".nycrc"))
+        os.symlink(Task.workspace / "out/.nycrc", compile()._1.path / ".nycrc")
+    }
+
+    def nycrcBuilder: Task[Path] = Task.Anon {
+      val compiled = compile()._1.path
+      val fileName = ".nycrc"
+      val config = compiled / fileName
+      val customConfig = Task.workspace / fileName
+
+      val content =
+        s"""|{
+            |  "report-dir": ${ujson.Str(s"${moduleDeps.head}_coverage")},
+            |  "extends": "@istanbuljs/nyc-config-typescript",
+            |  "all": true,
+            |  "include": ${ujson.Arr.from(coverageDirs())},
+            |  "exclude": ["node_modules", "*/**/*.test.ts"],
+            |  "reporter": ["text", "html"],
+            |  "require": ["ts-node/register", "tsconfig-paths/register"]
+            |}
+            |""".stripMargin
+
+      if (!os.exists(customConfig)) os.write.over(config, content)
+      else os.copy.over(customConfig, config)
+
+      os.write.over(config, content)
+
+      config
+    }
+
+    // web browser - serve coverage report
+    def htmlReport: T[Unit] = Task {
```

Comment:
Let's make this return a `T[PathRef]` containing the `.html` file or folder containing the report, in case downstream tasks want to use it